src/main/java/org/jabref/gui/fieldeditors/KeywordsEditor.java

```diff
@@ -1,24 +1,152 @@
+import javafx.scene.layout.HBox;
+
+import org.jabref.gui.ClipBoardManager;
+import org.jabref.gui.DialogService;
+import org.jabref.gui.JabRefDialogService;
+import org.jabref.gui.actions.ActionFactory;
+import org.jabref.gui.actions.SimpleCommand;
+import org.jabref.gui.actions.StandardActions;
 import org.jabref.gui.autocompleter.SuggestionProvider;
+import org.jabref.gui.keyboard.KeyBindingRepository;
+import org.jabref.gui.util.ViewModelListCellFactory;
 import org.jabref.logic.integrity.FieldCheckers;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.Keyword;
 import org.jabref.model.entry.field.Field;
 import org.jabref.preferences.PreferencesService;
 
-public class KeywordsEditor extends SimpleEditor implements FieldEditorFX {
+import com.airhacks.afterburner.views.ViewLoader;
+import com.dlsc.gemsfx.ChipView;
+import com.dlsc.gemsfx.TagsField;
+import jakarta.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KeywordsEditor extends HBox implements FieldEditorFX {
+    private static final Logger LOGGER = LoggerFactory.getLogger(KeywordsEditor.class);
+
+    @FXML private TagsField<Keyword> keywordTagsField;
+
+    @Inject private PreferencesService preferencesService;
+    @Inject private DialogService dialogService;
+    @Inject private UndoManager undoManager;
+    @Inject private ClipBoardManager clipBoardManager;
+    @Inject private KeyBindingRepository keyBindingRepository;
+
+    private final KeywordsEditorViewModel viewModel;
 
     public KeywordsEditor(Field field,
                           SuggestionProvider<?> suggestionProvider,
-                          FieldCheckers fieldCheckers,
-                          PreferencesService preferences,
-                          UndoManager undoManager) {
-        super(field, suggestionProvider, fieldCheckers, preferences, undoManager);
+                          FieldCheckers fieldCheckers) {
+
+        ViewLoader.view(this)
+                  .root(this)
+                  .load();
+
+        this.viewModel = new KeywordsEditorViewModel(
+                field,
+                suggestionProvider,
+                fieldCheckers,
+                preferencesService,
+                undoManager);
+
+        keywordTagsField.setCellFactory(new ViewModelListCellFactory<Keyword>().withText(Keyword::toString));
+        keywordTagsField.setTagViewFactory(this::createTag);
+
+        keywordTagsField.setSuggestionProvider(request -> viewModel.getSuggestions(request.getUserText()));
+        keywordTagsField.setConverter(viewModel.getStringConverter());
+        keywordTagsField.setMatcher((keyword, searchText) -> keyword.get().toLowerCase().startsWith(searchText.toLowerCase()));
+        keywordTagsField.setComparator(Comparator.comparing(Keyword::get));
+
+        keywordTagsField.setNewItemProducer(searchText -> viewModel.getStringConverter().fromString(searchText));
```

Comment:
I think there is no need for the `ItemProducer` and the `StringConverter` again, because I created a new Keyword with the requested text and added it to the beginning of the suggestion list in the `getSuggestions` method.